### PR TITLE
 Fix bug related to SoftDTWLossPytorch with option normalize=True when used on inputs of different lengths

### PR DESCRIPTION
**Pull request by YannCabanes**
This PR is related to the issue https://github.com/tslearn-team/tslearn/issues/473 reporting an error message when `SoftDTWLossPytorch`  with option `normalize=True` is used on inputs of different lengths.
The bug is coming from the use of `torch.cat` (https://pytorch.org/docs/stable/generated/torch.cat.html) which impose the lengths of the inputs to be equal. (The shape of the inputs is `[batch_size, length, dim]`.)
The use of `torch.cat` is removed in this PR.

### CODE REVIEW

**Comment by YannCabanes** [comment]
All checks have passed.

**Comment by YannCabanes** [comment]
All checks have passed.
